Remove unneeded & testing dependencies from recipe

fedarko opened the pull request:
I'm not sure this will fully resolve https://github.com/biocore/songbird/issues/122, but this should at least mitigate some of the errors.

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

fedarko committed:
Remove unneeded & testing dependencies from recipe

not sure this will fully resolve
https://github.com/biocore/songbird/issues/122
but this should at least mitigate some of the errors.

fedarko committed:
bump build number

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * License is not an SPDX identifier (or Other) nor an SPDX license expression.

fedarko committed:
Fix license name

Now it matches https://github.com/conda-forge/deicode-feedstock/blob/master/recipe/meta.yaml
at least?

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

fedarko commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.19.1, conda-smithy 3.6.14, and conda-forge-pinning 2020.03.20

mortonjt commented:
Looks good to me!

mortonjt commented:
Thanks @fedarko !!
